### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by kazumaki**

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+def bubble_sort(arr)
+  length = arr.length
+  for i in 0..length - 1 do
+    for j in 0..length-i-2 do
+      if arr[j] > arr[j + 1]
+        carry = arrToSort[j + 1]
```
Comment (wathigo) [inline bubble_sort.rb]: **[REQUIRED]**  `arrToSort` variable is not defined. After changing it to `arr` the code passes the sample test cases. Let's fix this

**Review by wathigo — CHANGES_REQUESTED** [review]
### Hi @kazumaki  and @adaorachi  :wave:
Great work on the bubble sort algorithm!
The `bubble_sort_by` method works well but we have a problem in the `bubble_sort` method.
Let's fix all the linter errors. 
See the comment :point_down:  

**After implementing the changes, submit another request for review.**

_Happy Coding :computer:  
Reviewed by: **Simon Wathigo**_

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      arr[j + 1] = arr[j]
+      arr[j] = carry
+    end
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  (0..arr.length - 2).each do |i|
+    (1..arr.length - (i + 1)).each do |j|
```
Comment (samgaco) [inline bubble_sort.rb:21]: - [ ] If no block is given **bubble_sort_by** should raise and error

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @kazumaki and @adaorachi :high_brightness: 

Just one thing left to change before proceeding. 

- [ ]  If no block is given bubble_sort_by should raise and error

fix that and submit back for another code review.

Regards!

----------------------------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: . However the project is not approved yet :red_circle: . check :arrow_down: 

- According to the Git Flow guidelines:
  - Create a development branch, set it as the default branch.
  - Set the target branch of this pull request (i.e the branch in which you want to merge) to be development instead of master.

### That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

**Ahmed Mahfoudh**

**Review by stratospherique — APPROVED** [review]
### Congrats :tada: Your project is approved :heavy_check_mark: And you may merge to master :+1: :+1: 

Keep up the good work :muscle: 
**Ahmed Mahfoudh**

**Pull request merged**